### ISSUE TITLE
Intel Iris Xe Graphics rendering issues with every 3D game under DXVK

### ISSUE THREAD

**Issue by ruineka** [doitsujin/dxvk#2374]
I haven't seen anyone mention anything regarding Intel and I don't see a way to ask questions or report issues that happen with virtually all 3D geometry under DXVK so I apologize if this isn't the correct way to do this.

All versions of Mesa up to bleeding edge were tested using all sorts of distributions on the OneXPlayer. The overall performance of the games are flawless and in many cases faster than Windows natively, however, assets flicker and disappear and reappear frequently. From what I gather it seems as though it's stuff that is rendered from a distance and it doesn't seem to happen as often with objects up close.

If you are interested in having proper bug reports with api traces for individual games, please let me know, I'd love to start helping out with this! If you aren't wanting to see Intel related reports yet that is fine too.

**Comment by K0bin** [comment]
DXVK works just fine on AMD and Nvidia GPUs so this is almost certainly a driver problem.

**Comment by ruineka** [comment]
> DXVK works just fine on AMD and Nvidia GPUs so this is almost certainly a driver problem.

That was what I was thinking, but I wasn't sure. For all I know Intel doesn't support an essential vulkan extension or something DXVK uses. Nobody is talking about Intel either so I'm walking on uncharted grounds with a very niche product. It doesn't seem to happen on native games under OpenGL (Tomb Raider) for example.

**Comment by Sigtechnica** [comment]
might be related to your issue (or could be you :) ) https://gitlab.freedesktop.org/mesa/mesa/-/issues/5662

**Comment by ruineka** [comment]
> might be related to your issue (or could be you :) ) https://gitlab.freedesktop.org/mesa/mesa/-/issues/5662

That's me, I reported it there as well just in case it was driver related and not DXVK.

**Comment by WinterSnowfall** [comment]
> . Nobody is talking about Intel either so I'm walking on uncharted grounds with a very niche product.

FWIW, I've been using an Iris Xe GPU (the i5-1135G7 80CU version) for a while now with DXVK to play several D3D9 games (Dragon Age - Origins, Majesty 2, Sins of a Solar Empire etc.) and I can't say I've run across any issues, certainly not the constant flickering you mention. Are you sure the issues you are seeing are not caused by whatever desktop compositor you are using?

**Comment by ruineka** [comment]
> > . Nobody is talking about Intel either so I'm walking on uncharted grounds with a very niche product.
> 
> FWIW, I've been using an Iris Xe GPU (the i5-1135G7 80CU version) for a while now with DXVK to play several D3D9 games (Dragon Age - Origins, Majesty 2, Sins of a Solar Empire etc.) and I can't say I've run across any issues, certainly not the constant flickering you mention. Are you sure the issues you are seeing are not caused by whatever desktop compositor you are using?

Hey Intel gamers do exist!! Thanks for the reply, I'm not sure if Ive tested any D3D9 games other than Sonic Adventure 2 Battle..That game didn't flicker. I'll compile a list of games that have these issues. Maybe we have a common dominator somewhere.

Ark Survival Evolved
Tekken 7
Elder Scrolls Online
Path of Exile (Rarely flickers but does)
Destroy All Humans
Tales of Arise
Dragon ball Z Kakarot

I can find a free to play game on Steam that flickers on my end if you could test it on yours as well. It's definitely not a compositor issue by the the way, unless all the distros I've used use the same one.

**Comment by WinterSnowfall** [comment]
>  I'm not sure if Ive tested any D3D9 games other than Sonic Adventure 2 Battle..That game didn't flicker.

Interesting... I for sure did not test any D3D11 games, since I don't expect the GPU's performance to be great with more recent games, but maybe give I'll take a UE4 engine-based title for a spin this weekend and report back in case I see any flickers. Perhaps it is indeed a Vulkan extension only used in this context which is borked in the anv implementation, who knows.

**Comment by Charmert** [comment]
> 

Hi, I have the same GPU, but when I try to start a game with Vulkan there is an infinite black screen or flickering exist. I am not a Linux user, just trying to use it with copying d3d files to games location on windows. Do I have to do something else except his?

**Comment by Charmert** [comment]
> > I'm not sure if Ive tested any D3D9 games other than Sonic Adventure 2 Battle..That game didn't flicker.
> 
> Interesting... I for sure did not test any D3D11 games, since I don't expect the GPU's performance to be great with more recent games, but maybe I'll take a UE4 engine-based title for a spin this weekend and report back in case I see any flickers. Perhaps it is indeed a Vulkan extension only used in this context which is borked in the anv implementation, who knows.
> 
> Edit: I've tried [Nomads of Driftland](https://store.steampowered.com/app/1400520/Nomads_of_Driftland/) (a D3D11 game) and I'm not seeing anything odd...



> 

Hi, I have the same GPU(iris 80 EU), but when I try to start a game with Vulkan there is an infinite black screen or flickering exist. I am not a Linux user, just trying to use it with copying d3d files to games location on windows. Do I have to do something else except his?

**Comment by Joshua-Ashton** [comment]
You have to tell Intel to make any form of effort with their Windows Vulkan driver first. It barely functions with anything.

**Comment by Charmert** [comment]
> You have to tell Intel to make any form of effort with their Windows Vulkan driver first. It barely functions with anything, so don't expect DXVK to work.

Yes, probably this caused the problem I guess. Because I tried the same game with another Windows laptop with Gtx 960m and worked like charm. I got a huge fps boost thanks to this API. On the other hand on Iris laptop, it is just black screens on any game sadly.
